lib/modules/manager/gradle/extract.ts

```diff
@@ -70,6 +76,13 @@ export async function extractAllPackageFiles(
       } else if (isTOMLFile(packageFile)) {
         const updatesFromCatalog = parseCatalog(packageFile, content);
         extractedDeps.push(...updatesFromCatalog);
+      } else if (isGcvLockFile(packageFile)) {
+        // Skip and do not trigger parsing, wait for VERSIONS_PROPS
+      } else if (isGcvPropsFile(packageFile)) {
+        if (usesGcv(packageFile, fileContents)) {
+          const updatesFromGcv = parseGcv(packageFile, fileContents);
+          extractedDeps.push(...updatesFromGcv);
+        } // else skip silently since this is just a name collision
```

Comment:
@janhoy , can you please merge `main` into your branch?
With PR https://github.com/renovatebot/renovate/pull/19232 in place, the no-op check for `isGcvLockFile()` shouldn't be necessary anymore - `versions.lock` will simply be ignored as it has no handler.

Related to that, please combine the check `isGcvPropsFile` and `usesGcv` into one if-clause. A _name collision_ with nuget-specific files, should also fall through now.

Reply:
Done. Nice simplification.